mempool/txpool/legacypool/legacypool.go

```diff
@@ -1437,6 +1467,12 @@ func (pool *LegacyPool) promoteExecutables(accounts []common.Address) []*types.T
 	// Track the promoted transactions to broadcast them at once
 	var promoted []*types.Transaction
 
+	// create a checker instance to ensure all queued txs are valid
+	var checker func(tx *types.Transaction) error
```

Comment:
This variable doesn't seem necessary...we can just construct this where it's used can't we?